### PR TITLE
Handling fixed column relationships by `specific_combinations` and `SpecificCombinationTransformer`.

### PR DESCRIPTION
**Pull request by jalr4ever**
## Description

I haved added a new `SpecificCombinationTransformer`, and modifed `FixedCombinationTransformer`. And a test case for `SpecificCombinationTransformer`.

## Motivation and Context

`SpecificCombinationTransformer` would transforming when user specified some column-mapping combination, and `FixedCombinationTransformer` would not running at the same time.

I make `FixedCombinationTransformer` as default column-mapping combination detector and transformer.As for `SpecificCombinationTransformer`, it would transforming when user specified some column-mapping combinations, make sure the sample data column-mapping adhere to the specified combinations. At the same time,  `FixedCombinationTransformer` would not run transforming when `SpecificCombinationTransformer` activated.

## How has this been tested?

See test case.

## Types of changes

- [ ] Maintenance (no change in code, maintain the project's CI, docs, etc.)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: sdgx/data_processors/transformers/fixed_combination.py [hunk 1]
```diff
@@ -15,41 +15,90 @@ class FixedCombinationTransformer(Transformer):
...
+
+        # Check if exist fixed combinations, if not exist, needn't run this auto-detect transform.
         self.fixed_combinations = metadata.get("fixed_combinations")
+        if self.fixed_combinations is not None and len(self.fixed_combinations) > 0:
+            logger.info(
+                "Fit data using FixedCombinationTransformer(not existed)... Finished (No action)."
+            )
+            self.is_exist_fixed_combinations = False
+            self.fitted = True
+            return
```
Comment (Wh1isper) [inline sdgx/data_processors/transformers/fixed_combination.py:99]: Equal to these?

```suggestion
        # Check if exist specific combinations by user. If True, needn't run this auto-detect transform.
        if metadata.get("specific_combinations"):
            logger.info(
                "Fit data using FixedCombinationTransformer(been specified)... Finished (No action)."
            )
            self.is_been_specified = True
            self.fitted = True
            return

        # Check if exist fixed combinations, if not exist, needn't run this auto-detect transform.
        if  metadata.get("fixed_combinations"):
            logger.info(
                "Fit data using FixedCombinationTransformer(not existed)... Finished (No action)."
            )
            self.is_exist_fixed_combinations = False
            self.fitted = True
            return
```

**Comment by Wh1isper** [comment]
Do we really need to distinguish between `is_been_specified ` and `is_exist_fixed_combinations`, seems just for log... Or do you think it would be beneficial downstream?

**Comment by jalr4ever** [comment]
> Do we really need to distinguish between `is_been_specified ` and `is_exist_fixed_combinations`, seems just for log... Or do you think it would be beneficial downstream?

@Wh1isper Yes we need. 

Currently, the consideration is that when the shape of the data does not meet expectations, it can be determined from the logs what caused the lack of transformation; on the other hand, there may be other combinations in the future, and different scenarios may have different logic.

**Comment by Wh1isper** [comment]
@jalr4ever Ok and I suggest replace `is_exist_fixed_combinations` with `@property` decorator

Like this:

```python
@property
def is_exist_fixed_combinations(self) -> bool:
  return bool(self.is_exist_fixed_combinations)
```

Then we can reduce the code in `fit` function, no need tow `if` block but one is enough

File: sdgx/data_processors/transformers/fixed_combination.py [hunk 2]
```diff
@@ -15,41 +15,89 @@ class FixedCombinationTransformer(Transformer):
...
+        if metadata.get("specific_combinations"):
+            logger.info(
+                "Fit data using FixedCombinationTransformer(been specified)... Finished (No action)."
+            )
+            self.is_been_specified = True
+            self.fitted = True
+            return
+
+        # Check if exist fixed combinations, if not exist, needn't run this auto-detect transform.
         self.fixed_combinations = metadata.get("fixed_combinations")
```
Comment (Wh1isper) [inline sdgx/data_processors/transformers/fixed_combination.py:92]:
```suggestion
        self.fixed_combinations = metadata.get("fixed_combinations") or dict()
```

as `self.fixed_combinations: dict[str, set[str]]`

**Review by Wh1isper — APPROVED** [review]
LGTM

**Pull request merged**